pkg/reconciler/buildrun/buildrun.go

```diff
@@ -123,6 +125,14 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 				buildRun.Labels = make(map[string]string)
 			}
 
+			// make sure the BuildRun has not already been cancelled
+			if buildRun.IsCanceled() {
```

Comment:
Good catch, a BuildRun that is canceled even before a TaskRun is created. :-)

Reply:
:-)